components/servo/tests/multiprocess.rs

```diff
@@ -59,12 +62,24 @@ fn main() {
         return run_content_process(token);
     }
 
-    run_api_tests!(
-        setup: |builder| {
-            let mut opts = opts::Opts::default();
-            opts.multiprocess = true;
-            builder.opts(opts)
-        },
-        test_multiprocess_preference_observer
-    );
+    // See nextest custom test harness requirements:
+    // <https://nexte.st/docs/design/custom-test-harnesses/>
+    let args = std::env::args().collect::<Vec<_>>();
+
+    // If there are no ignored tests or if the test harness doesn't support ignored tests,
+    // the output MUST be empty
+    if args.contains(&String::from("--ignored")) {
+        return;
+    }
+
+    // Expected output:
+    // ```
+    // my-test-1: test
+    // my-test-2: test
+    // ```
+    if args.contains(&String::from("--list")) {
+        println!("test_multiprocess_preference_observer: test")
```

Comment:
I think we would want to exit here (and not run the test), otherwise an error in the test, would cause the test list gathering step to also fail.